mp/src/game/client/momentum/ui/HUD/hud_speedometer_label.cpp

```diff
@@ -0,0 +1,285 @@
+        if (m_bSupportsEnergyUnits)
+        {
+            const auto gravity = m_cvarGravity.GetFloat();
+            const auto pPlayer = C_MomentumPlayer::GetLocalMomPlayer();
+            Vector velocity = pPlayer->GetAbsVelocity();
+            const auto pEntData = pPlayer->GetCurrentUIEntData();
+            if (!pEntData)
+                break;
+
+            m_flCurrentValue =
+                (velocity.LengthSqr() / 2.0f + gravity * (pPlayer->GetLocalOrigin().z - pEntData->m_flLastJumpZPos)) /
+                gravity;
+        }
+        break;
+    default:
+        break;
+    }
+}
+
+void SpeedometerLabel::Colorize()
+{
+    if (m_flNextUpdateCheck > gpGlobals->curtime)
+        return;
+
+    switch (m_eColorizeType)
+    {
+    case SPEEDOMETER_COLORIZE_RANGE:
+        ColorizeRange();
+        break;
+    case SPEEDOMETER_COLORIZE_COMPARISON:
+        ColorizeComparison();
+        break;
+    case SPEEDOMETER_COLORIZE_COMPARISON_SEPARATE:
+        ColorizeComparisonSeparate();
+        break;
+    default:
+        SetFgColor(m_NormalColor);
+        break;
+    }
+
+    m_flPastValue = m_flCurrentValue;
+    m_flNextUpdateCheck = gpGlobals->curtime + MOM_COLORIZATION_CHECK_FREQUENCY;
+}
+
+void SpeedometerLabel::SetFadeOutAnimation(char *animationName, float *animationAlpha)
+{
+    m_pSzAnimationName = animationName;
+    m_pFlAlpha = animationAlpha;
+}
+
+void SpeedometerLabel::ClearFadeOutAnimation()
+{
+    m_pSzAnimationName = nullptr;
+    m_pFlAlpha = nullptr;
+}
+
+bool SpeedometerLabel::StartFadeout()
+{
+    Assert(HasFadeOutAnimation());
+    g_pClientMode->GetViewportAnimationController()->StopAnimationSequence(GetParent(), m_pSzAnimationName);
+    *m_pFlAlpha = 255.0f;
+    return g_pClientMode->GetViewportAnimationController()->StartAnimationSequence(m_pSzAnimationName);
+}
+
+void SpeedometerLabel::SetRangeList(RangeList *pRangeList)
+{
+    if (m_pRangeList)
+        delete m_pRangeList;
+    m_pRangeList = pRangeList;
+}
+
+void SpeedometerLabel::ColorizeRange()
+{
+    int roundedCurrentValue = RoundFloatToInt(m_flCurrentValue);
+    Color newColor = m_NormalColor;
+    if (m_pRangeList)
+    {
+        for (auto i = 0; i < m_pRangeList->Count(); i++)
+        {
+            Range_t range = (*m_pRangeList)[i];
+            if (roundedCurrentValue >= range.min && roundedCurrentValue <= range.max)
+            {
+                newColor = range.color;
+                break;
+            }
+        }
+    }
+    SetFgColor(newColor);
+}
+
+void SpeedometerLabel::ColorizeComparison()
+{
+    if (m_eUnitType == SPEEDOMETER_UNITS_ENERGY && m_bSupportsEnergyUnits)
+    {
+        m_flDiff = m_flCurrentValue - m_flPastValue;
+    }
+    else
+    {
+        m_flDiff = fabs(m_flCurrentValue) - fabs(m_flPastValue);
+    }
+    SetFgColor(
+        MomUtil::GetColorFromVariation(m_flDiff, COLORIZE_DEADZONE, m_NormalColor, m_IncreaseColor, m_DecreaseColor));
```

Comment:
Can be same line

Reply:
Stupid visual studio.. must've happened when I copy pasted the code into it's own function